### PR TITLE
Add amp-ad support for videointelligence

### PR DESCRIPTION
**Pull request by cosmin-vi**
Adding videointelligence ad network to the list of supported ad networks.

We signed the CLA (company name: video intelligence).

### CODE REVIEW

File: ads/_config.js [hunk 1]
```diff
@@ -899,6 +899,13 @@ export const adConfig = {
     renderStartImplemented: true,
   },
 
+  'videointelligence': {
+    preconnect: [
```
Comment (zhouyx) [inline ads/_config.js]: nit: No need for an array if there's only single item.

File: ads/videointelligence.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/videointelligence.js]: nit: 2018

File: ads/videointelligence.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<!---
+Copyright 2017 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/videointelligence.md]: nit: 2018

File: ads/videointelligence.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+import {loadScript, validateData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function videointelligence(global, data) {
+
+  validateData(data, ['publisherId', 'channelId']);
```
Comment (zhouyx) [inline ads/videointelligence.js]: FYI: You can put `['placementId', 'iabCategory', 'language']` as the third param. It throws async user errors for unknown attributes other than these three. 
It's an optional param, up to you.

File: ads/_config.js [hunk 5]
```diff
@@ -899,6 +899,13 @@ export const adConfig = {
...
+      'https://s.vi-serve.com',
+    ],
+    renderStartImplemented: true,
```
Comment (zhouyx) [inline ads/_config.js]: I noticed you indicate that you support the [renderStart API](https://github.com/ampproject/amphtml/blob/master/ads/README.md#available-apis). However I couldn't receive the `render-start` message when testing with your example. Could you double check your implementation? 
We highly recommend supporting the API for better user experience if you haven't done so.

File: ads/videointelligence.js (hunk 4 above)
Comment (cosmin-vi) [reply ads/videointelligence.js]: Thank you very much for the suggestion, but we'd like to keep it without optional parameters specified.

File: ads/_config.js (hunk 5 above)
Comment (cosmin-vi) [reply ads/_config.js]: We checked our implementation and we were indeed not calling window.context.renderStart() 
This should be fixed now.
Thank you.

**Comment by zhouyx** [comment]
@rsimha Do you have any idea why the unit test is failing here? Seems to be unrelated. I've retried, no luck.

**Comment by rsimha** [comment]
@zhouyx No idea. You could try syncing forward, in case the tests were broken on `master` at the branch point commit. If not, I'd try asking whoever wrote those tests.

**Comment by zhouyx** [comment]
@cosmin-vi As @rsimha suggested, could you try syncing forward. 
I suspect that this is due to a recent change to `--local-changes` that include the flakey test suddenly.  I've created #17137 to fix those tests.

**Comment by cosmin-vi** [comment]
@zhouyx what do you mean by syncing forward ? rebasing ?

**Comment by zhouyx** [comment]
Yes. Try rebasing to latest master

**Comment by cosmin-vi** [comment]
Didn't help. Now the integration tests failed as well :(

**Comment by cosmin-vi** [comment]
Hi @zhouyx 
Do you have any other suggestion ? I see there is no progress on #17137

**Comment by zhouyx** [comment]
Unfortunately we haven't fixed the test failure yet. I agree your change is not related to the test failure, I'll ask for a force merge for this PR. Thanks. 

to @erwinmombay Could you help force merge the PR? Thank you!

**Comment by cosmin-vi** [comment]
Thank you very much @zhouyx and @lannka  :+1:

**Pull request merged**